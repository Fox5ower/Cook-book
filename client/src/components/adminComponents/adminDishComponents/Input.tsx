import React, { Component, ChangeEvent } from 'react';

interface MyProps {
    name: string,
    maxLength: number,
    value?: string | Array<string>,
    onChange?: any | null
}

class Input extends Component<MyProps> {

    constructor(props: MyProps) {
        super(props)

    }

    capitalizeFirstLetter(string: string) {
        return string.charAt(0).toUpperCase() + string.slice(1);
    }

    render() {
        return (
            <div className="input-container">
                <label htmlFor={this.props.name}>{this.capitalizeFirstLetter(this.props.name)}</label>
                <textarea maxLength={this.props.maxLength} name={this.props.name} id={this.props.name} value={this.props.value} onChange={e => this.props.onChange(e)} />
            </div>
        )
    }
}


export default Input;
