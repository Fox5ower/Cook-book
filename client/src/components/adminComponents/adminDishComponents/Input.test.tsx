import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import Input from './Input'

const fakeProps = {
  name: 'Test name',
  placeholder: 'Test Placeholder',
  maxLength: 12,
  value: 'Test Value',
}

it('Should render Recipes preview', () => {
  const { getByLabelText } = render(
    <Input
      name={fakeProps.name}
      placeholder={fakeProps.placeholder}
      maxLength={fakeProps.maxLength}
      value={fakeProps.value}
    />
  )

  const placeholderText = getByLabelText(/Test Placeholder/i)
  expect(placeholderText).toBeInTheDocument()
})

it('Should match the snapshot', () => {
  const tree = renderer
    .create(
      <Input
        name={fakeProps.name}
        placeholder={fakeProps.placeholder}
        maxLength={fakeProps.maxLength}
        value={fakeProps.value}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
