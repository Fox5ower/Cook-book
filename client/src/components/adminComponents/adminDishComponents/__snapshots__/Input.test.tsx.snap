// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="input-container"
>
  <textarea
    id="Test name"
    maxLength={12}
    name="Test name"
    onChange={[Function]}
    required={true}
    value="Test Value"
  />
  <label
    className="label"
    htmlFor="Test name"
  >
    Test Placeholder
  </label>
</div>
`;
